tensorflow/contrib/learn/python/learn/estimators/estimator.py

```diff
@@ -1203,12 +1211,6 @@ def _get_eval_ops(self, features, labels, metrics):
     model_fn_ops = self._call_model_fn(
         features, labels, model_fn_lib.ModeKeys.EVAL)
```

Comment:
Thanks for your review. 

Now `feature_engineerng_fn(features, label)` is called by `Estimator.eval()` twice.
Because `features` can be a `dict` which is passed by pointer for function, hence `features` will be converted by `feature_engineering_fn` twice. It's dangerous and might get in trouble.

So we should assure that `feature_engineering_fn` only be called once for `train`, `eval` and `predict`. That's why I send metrics to `call_model_fn`. 

In fact, I have added an unit test below to catch the error.